File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.cpp
class Solution {
public:
    bool checkDivisibility(int n) {
        int sum=0;
        int product=1;
        int x=n;

        // Har digit ko nikalenge
        while(x!=0){
            int digit=x%10;
            x/=10;
            sum+=digit;
            product*=digit;
        }
        // n ko digit sum + digit product se divide karo
        return n%(sum+product)==0;
    }
};

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.js
/**
 * @param {number} n
 * @return {boolean}
 */
var checkDivisibility = function(n) {
    let sum=0;
    let product=1;
    let x=n;

    // Har digit ko nikalenge
    while(x!=0){
        let digit=x%10;
        x=Math.floor(x/10);

        sum+=digit;
        product*=digit;
    }
    // Divisibility check
    return n%(sum+product)===0;
};